src/llmcompressor/entrypoints/utils.py

```diff
@@ -79,20 +84,29 @@ def post_process(
     Raises:
         ValueError: If saving fails due to an invalid `output_dir` or other issues.
     """
-    if output_dir is not None:
+    if model_args is not None and output_dir is not None:
+        if recipe_args is not None and getattr(recipe_args, "stage", None) is not None:
+            output_dir = os.path.join(output_dir, recipe_args.stage)
```

Comment:
if the user only has one stage, does it still get saved in output_dir / stage?

Reply:
no, only if `recipe_args` has stage attr. That is passed in as oneshot(..., stage="foo-stage").